source/agora/consensus/state/UTXOSet.d

```diff
@@ -82,11 +83,23 @@ abstract class UTXOCache
         {
             auto utxo_hash = UTXO.getHash(tx_hash, idx);
 
-            // Melting UTXOs will melt after 2016 blocks except ones that go to
-            // the CommonsBudget and others will melt at the next block.
-            auto utxo_value = UTXO((melting && output.address != commons_budget) ?
-                height + 2016 : height + 1, tx.type, output);
-            this.add(utxo_hash, utxo_value);
+            // In general, melting UTXOs will be available after 2016 blocks
+            //
+            // UTXOs which are towards the Commons Budget are excluded from this
+            // rule, and will be available from the next block.
+            if (melting && output.address != commons_budget)
+                this.add(utxo_hash, UTXO(height + 2016, tx.type, output));
+
+            // Additionally, if a transaction has two or more outputs,
+            // the first one is allowed to be under the minimum freezing amount.
```

Comment:
Is there any reason the amount only under `Amount.MinFreezeAmount` is unfrozen at the index 0?

Reply:
It would be (by convention) the refund transaction